docs_src/advanced/binders/msgspec/tests.py

```diff
@@ -0,0 +1,141 @@
+from typing import Any, Dict, List
+
+from msgspec import Struct
+
+from xpresso import App, Path
+from xpresso.testclient import TestClient
+
+from .functions import FromJson
+
+
+class Item(Struct):
+    foo: str
+    bar: int
+
+
+async def echo_item(items: FromJson[List[Item]]) -> int:
```

Comment:
I can see how the backend used for parsing an input type/generating JSON Schema is dispatchable based on a `FromJson` annotation - can the same be done for the return type? E.g. how would you return a `msgspec.Struct` type from a handler and have everything still work?

Reply:
Great question!

We're unfortunately a bit tied to Pydantic for responses, that part isn't as pluggable as I'd like. I'll have to think a bit more about how we could make that more pluggable, it should be possible. I think we may want to have some sort of `-> JsonResponse[MyMsgSpecStruct]` where `JsonResponse ` somehow tells Xpresso both how to serialize the response and how to build the OpenAPI schema.

Reply:
@jcrist is msgspec _much_ faster at encoding something like `list[Struct]` into bytes than it would be to convert it to a `list[dict]` and then run that through a fast json serializer (like orjson)? I think the `list[Struct]` -> `list[dict]` part doesn't currently exist, so I guess this is a theoretical question.